src/objective/quantile_obj.cu

```diff
@@ -80,15 +79,8 @@ class QuantileRegression : public ObjFunction {
 
     linalg::ElementWiseKernel(
         ctx_, gpair, [=] XGBOOST_DEVICE(std::size_t i, GradientPair const&) mutable {
-          auto idx = linalg::UnravelIndex(static_cast<std::size_t>(i),
-                                          {static_cast<std::size_t>(n_samples),
-                                           static_cast<std::size_t>(alpha.size()),
-                                           static_cast<std::size_t>(n_targets / alpha.size())});
-
-          // std::tie is not available for cuda kernel.
-          std::size_t sample_id = std::get<0>(idx);
-          std::size_t quantile_id = std::get<1>(idx);
-          std::size_t target_id = std::get<2>(idx);
+          auto [sample_id, quantile_id, target_id] =
+              linalg::UnravelIndex(i, n_samples, alpha.size(), n_targets / alpha.size());
```

Comment:
@hcho3 I think your c++-17 work is really helpful.

Reply:
Awesome!